code/obj/item/grenades.dm

```diff
@@ -981,13 +995,58 @@ PIPE BOMBS + CONSTRUCTION
 
 		else if (iscuttingtool(A) && !(src.slashed) && (src.primed)) // cutting open a lit firework is a bad idea!
 			boutput(user, "<span class='alert'>You cut open [src], but the lit primer ignites the contents!</span>")
-			boom()
+			boom(user)
 			return
 
 		else if (iscuttingtool(A) && (src.slashed))
 			boutput(user, "[src] has already been cut open and emptied.")
 			return
 
+/obj/item/firework/bootleg
+	name = "bootleg firework"
+	desc = "A consumer-grade pyrotechnic, often used in celebrations. This one seems to be missing a label, weird."
+
+	New()
+		..()
+		create_reagents(10)
+
+		if (prob(30))
+			reagents.add_reagent("flashpowder", 5) // must've been a mix-up!
+
+			if (prob(15))
+				reagents.add_reagent("blackpowder", 5) // thats one hell of a mix-up
+				src.bootleg_level = 2
+			else
+				reagents.add_reagent("flashpowder", 5) // this way every firework has 10u reagents
+				src.bootleg_level = 1
+
+		else
+			reagents.add_reagent("magnesium", 10)
+
+	afterattack(atom/target as mob|obj|turf|area, mob/user as mob)
+		if (src.bootleg_level > 0)
+			boutput(user, "<span class='alert'>You try to prime the firework, but the contents ignite immediately!</span>")
```

Comment:
Shouldn't this be something like "you accidentally prime the firework"? Because `afterattack()` is for when you attack something with `src` (so, the firework) inhand.

Reply:
fixed text to reflect accidental priming, also changed parent afterattack() text to match